### PR TITLE
Don't process Superscript inside Code, CodeBlock

### PR DESCRIPTION
**Pull request by scria1000**
Resolves issue #742

### CODE REVIEW

File: app/src/main/java/ml/docilealligator/infinityforreddit/utils/Utils.java [hunk 1]
```diff
@@ -88,64 +90,116 @@ public static String modifyMarkdown(String markdown) {
...
                 nCarets--;
+                i += 5;
+                offset += 5;
+            } else if (hasCodeRanges && nCarets > 0 && exists(codeRanges, offset, i)) {
+                for (int j = 0; j < nCarets; j++) {
+                    regexed.insert(i, "</sup>");
+                    i += 6;
+                    offset += 6;
+                }
+                nCarets = 0;
```
Comment (RSBat) [inline app/src/main/java/ml/docilealligator/infinityforreddit/utils/Utils.java:162]: This doesn't work for ``a^(one `two` three)``. `three` should be superscripted.

Comment (scria1000) [reply app/src/main/java/ml/docilealligator/infinityforreddit/utils/Utils.java:162]: Thanks!

**Comment by scria1000** [comment]
The caret parser probably needs an overhaul as well. Nested superscripts  `^(one^(two))` and `^(` unclosed superscripts are handled incorrectly for example.

**Comment by RSBat** [comment]
Yeah, another broken example:
```
a^(new \(
line) next
```

File: app/src/main/java/ml/docilealligator/infinityforreddit/utils/Utils.java (hunk 1 above)
Comment (RSBat) [reply app/src/main/java/ml/docilealligator/infinityforreddit/utils/Utils.java]: I did a bit more testing:
``a^one`two`three`` - three should be superscripted
``a^one`two three`four`` - four is superscripted only on new Reddit. On old Reddit space stops the superscript and there is no code block

**Comment by scria1000** [comment]
Output seems to be identical to the official app?

File: app/src/main/java/ml/docilealligator/infinityforreddit/utils/Utils.java (hunk 1 above)
Comment (scria1000) [reply app/src/main/java/ml/docilealligator/infinityforreddit/utils/Utils.java]: Thank you, will look into these.

**Comment by RSBat** [comment]
Sorry, its closing bracket in the middle
```
a^(new \)
line) next
```

Also I already forgot why I included line break, it doesn't work even in a single line

**Pull request closed without merging**